src/components/plots/xyPlot.ts

```diff
@@ -71,6 +71,49 @@ module Plottable {
       return this;
     }
 
+    protected _updateExtentsForAttr(attr: string) {
+      if (attr === "x") {
+        return super._updateExtentsForAttr(attr, this._xFilter());
+      } else if (attr === "y") {
+        return super._updateExtentsForAttr(attr, this._yFilter());
+      }
+      return super._updateExtentsForAttr(attr);
+    }
+
+    protected _xFilter(): _Accessor {
+      if (this._autoAdjustXScaleDomain) {
+        var yBinding = this._attrBindings.get("y");
+        if (yBinding != null) {
+          var yAccessor = yBinding.accessor;
+          var yScale = yBinding.scale;
+          if (yScale != null) {
+            return (d, i, dataset, metadata) => {
+              var range = yScale.range();
+              return Utils.Methods.inRange(yScale.scale(yAccessor(d, i, dataset, metadata)), range[0], range[1]);
+            };
+          }
+        }
+      }
+      return null;
+    }
+
+    protected _yFilter(): _Accessor {
+      if (this._autoAdjustYScaleDomain) {
+        var xBinding = this._attrBindings.get("x");
+        if (xBinding != null) {
+          var xAccessor = xBinding.accessor;
+          var xScale = xBinding.scale;
+          if (xScale != null) {
+            return (d, i, dataset, metadata) => {
+              var range = xScale.range();
+              return Utils.Methods.inRange(xScale.scale(xAccessor(d, i, dataset, metadata)), range[0], range[1]);
+            };
+          }
+        }
+      }
+      return null;
+    }
```

Comment:
(I assume that you're referring to the fact that `_xFilter()` and `_yFilter()` look similar.) I think that was the mistake we made with `normalizeDatasets():. Reluctant to repeat code, instead we added an obscure-looking method whose logic is hard to follow. The logic only applies to X being filtered based on Y, and Y being filtered based on X, so I don't think it's worth being generalized.